### PR TITLE
175 tests for createtimesheetentryform

### PR DESCRIPTION
**Pull request by ronikeuru**
Added tests for create timesheet entry form

### CODE REVIEW

File: components/form/TimesheetEntryForm.tsx [hunk 1]
```diff
@@ -55,21 +56,35 @@ const validateTimesheetEntryFields = (
     fields: TimesheetEntryFields
 ): TimesheetEntry => {
     const { timesheet, quantity, timeCategory, date, task } = fields
-    if (timesheet && quantity && date && timeCategory && task) {
-        return { ...fields, timesheet, quantity, date, timeCategory, task }
+    if (timesheet === undefined || timesheet === null) {
```
Comment (miikahyttinen) [inline components/form/TimesheetEntryForm.tsx:59]: Oisko `if (!timesheet)`

File: components/form/TimesheetEntryForm.tsx [hunk 2]
```diff
@@ -84,6 +99,9 @@ function TimesheetEntryForm({
         timesheetEntryFields.quantity?.toString() || ""
     )
 
+    const [errorMessage, setErrorMessage] = useState<string>("")
+    const errorHandler = (error: Error) => setErrorMessage(`${error}`)
```
Comment (miikahyttinen) [inline components/form/TimesheetEntryForm.tsx:103]: Onko jotain `error.toString()` juttua tms?

File: components/form/TimesheetEntryForm.tsx [hunk 3]
```diff
@@ -122,8 +140,14 @@ function TimesheetEntryForm({
     }
 
     const handleSubmit = () => {
-        const entry = validateTimesheetEntryFields(timesheetEntryFields)
-        onSubmit(entry)
+        try {
+            const entry = validateTimesheetEntryFields(timesheetEntryFields)
+            onSubmit(entry)
+        } catch (error) {
```
Comment (miikahyttinen) [inline components/form/TimesheetEntryForm.tsx:146]: Is it possible that `catch` catches something else than an actual error? Could this be typed then?

File: components/form/TimesheetEntryForm.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ import React, { useContext, useEffect, useState } from "react"
 import ErrorAlert from "../common/ErrorAlert"
 import { FromButtons } from "../common/FormFields"
 import { timesheetEntryFieldMetadata } from "@/lib/types/typeMetadata"
+import { isError } from "lodash"
 
 type TimesheetEntryFormBaseProps = FormBase<TimesheetEntry> & {
     entryOrNull?: TimesheetEntry
```
Comment (miikahyttinen) [inline components/form/TimesheetEntryForm.tsx:25]: No in delta but would it be enough to implicate possible nullness just with `?`

File: components/form/TimesheetEntryForm.tsx (hunk 3 above)
Comment (ronikeuru) [reply components/form/TimesheetEntryForm.tsx:146]: No. It catches errors that have been thrown in validateTimesheetEntryFields function. JS `try...catch` is just shit and you have to validate the error/exception type by yourself inside the catch. It's even in the [try...catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch-blocks) documentations. :/

**Review by miikahyttinen — APPROVED** [review]

**Pull request merged**